### PR TITLE
doc: make it explicit that `pre_present_notify` is supported on Wayland only

### PR DESCRIPTION
**Pull request by filnet**
fixes https://github.com/rust-windowing/winit/issues/3703

- [ ] Tested on all platforms changed
- [ ] Added an entry to the `changelog` module if knowledge of this change could be valuable to users
- [X] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [ ] Created or updated an example program if it would help users understand this functionality
- [ ] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

File: src/window.rs [hunk 1]
```diff
@@ -632,7 +632,8 @@ impl Window {
     ///
     /// ## Platform-specific
     ///
-    /// **Wayland:** - schedules a frame callback to throttle [`WindowEvent::RedrawRequested`].
+    /// - **Android / iOS / X11 / Web / Windows:** Unsupported.
```
Comment (filnet) [inline src/window.rs:635]: Should it say "Does nothing", "Not implemented" or "Unsupported" ?

Comment (madsmtm) [reply src/window.rs:635]: I think "Does nothing"

Comment (kchibisov) [reply src/window.rs:635]: `Unsupported` is fine, because we can add something in there later. 

you should also mention all the platforms we support.

Comment (filnet) [reply src/window.rs:635]: > you should also mention all the platforms we support.

`/// - **Wayland:** Schedules a frame callback to throttle [`WindowEvent::RedrawRequested`].`

This is not enough ?

Comment (kchibisov) [reply src/window.rs:635]: You miss `Orbital` and `macOS` in `Unsupported` list.

**Review by kchibisov — APPROVED** [review]

**Comment by kchibisov** [comment]
Thanks

**Pull request merged**